File: react-native/core/src/FerrostarCoreException.ts
class FerrostarCoreException extends Error {
  constructor(message: string) {
    super(message);
  }
}

export class NoResponseBodyException extends FerrostarCoreException {
  constructor() {
    super('Route request was successful but had no body.');
  }
}

export class NoRequestBodyException extends FerrostarCoreException {
  constructor() {
    super('Route request was not successful and had no request body.');
  }
}

export class InvalidStatusCodeException extends FerrostarCoreException {
  constructor(statusCode: number) {
    super(
      `Route request was not successful and had status code ${statusCode}.`
    );
  }
}

File: android/core/src/main/java/com/stadiamaps/ferrostar/core/FerrostarCoreException.kt
package com.stadiamaps.ferrostar.core

open class FerrostarCoreException : Exception {
  constructor(message: String) : super(message)

  constructor(message: String, cause: Throwable) : super(message, cause)

  constructor(cause: Throwable) : super(cause)
}

class InvalidStatusCodeException(val statusCode: Int) :
    FerrostarCoreException("Route request failed with status code $statusCode")

class NoCachedSession :
    FerrostarCoreException(message = "A resumable cached session snapshot was not found")

class NoResponseBodyException :
    FerrostarCoreException("Route request was successful but had no body bytes")

class UserLocationUnknown :
    FerrostarCoreException(
        "The user location is unknown; ensure the location provider is properly configured")
